### ISSUE TITLE
No more than one day during long run tests. system down

### ISSUE THREAD

**Issue by hagsac** [stereolabs/zed-unity#225]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues.
- [X] This issue is not a question, feature request, or anything other than a bug report directly related to this project.

### Description

It's a simple durability test. However, Unity is currently shutting down due to the following symptoms.
I can't find any reason.
The following phenomena and sometimes the left and right cameras are reversed or shaken.
The problem is serious.
zed2i is industrial and I plan to apply it to industrial sites.
My goal is to have the camera in operation for at least a year.
But even 24 hours is a failure.

What the hell is the problem? The reliability of this product is getting less and less. Is it true that it was made for industrial use?

There are many issues, but we are only reporting the most serious ones right now.
Currently, the zed2 series has a lot of huge problems.

https://user-images.githubusercontent.com/75370807/153035316-49bfe412-bc14-4548-93a9-f75422b1775b.mp4

### Steps to Reproduce

1. zed 4.6.4 install,    unity plugin install.
2. Add zed unity example
3. Simple time and sensor data visualization
No more programming in Unity

### Expected Result

365 days of trouble-free runtime

### Actual Result

The program is terminated without any reason and the camera cannot be connected.
Restart is possible only after disconnecting and reconnecting the USB cable to connect the camera.

### ZED Camera model

ZED2i

### Environment

```shell
windows 10 pro
gtx 1660 ti
sdk 3.6.4
unity 2020
```

**Comment by Bvallon-sl** [comment]
Hi,

I'm sorry to hear that.

Could you send the crash report generated by Unity at support@stereolabs.com ?
We will investigate the issue and come back with a solution as fast as possible.

Best regards,

**Comment by hagsac** [comment]
> Hi,
> 
> I'm sorry to hear that.
> 
> Could you send the crash report generated by Unity at [support@stereolabs.com](mailto:support@stereolabs.com) ? We will investigate the issue and come back with a solution as fast as possible.
> 
> Best regards,

sorry. I'm not a Unity expert.
Do I need to generate a crash report separately?
Or is it somewhere inside the build folder?
First of all, the error content did not pop up by default.

Excuse me, could you please briefly explain how I can view the crash report?

**Comment by hagsac** [comment]
> Hi,
> 
> I'm sorry to hear that.
> 
> Could you send the crash report generated by Unity at [support@stereolabs.com](mailto:support@stereolabs.com) ? We will investigate the issue and come back with a solution as fast as possible.
> 
> Best regards,


Playing in editor mode, I found the following error log:




Unhandled exception: Protocol error - failed to read magic number (error -2147483644, transferred 0/4)

Quitting shader compiler process

**Comment by Bvallon-sl** [comment]
Yes of course.

Unity should have generated a crash report automatically. It is available at :

C:\Users\USER\AppData\Local\Temp\Unity\Editor\Crashes (on windows)

best,

**Comment by hagsac** [comment]
> 
I mean, it asks if there is also a log of the application that has completed the build.

And at your request, I played it in editor mode and wrote it down in the comments above. Is that not enough?


-------------------
**Unhandled exception: Protocol error - failed to read magic number (error -2147483644, transferred 0/4)

Quitting shader compiler process**

**Comment by Bvallon-sl** [comment]
Unfortunately, no it is not enough as this error is Unity-related, it does not indicate at all what were wrong in the SDK.
